### ISSUE TITLE
Mastodon Multiple Instances

### ISSUE THREAD

**Issue by alexnya** [tootsuite/mastodon#7972]
This is not an issue more like a general question. I've been trying to find an answer for this but can you have multiple Mastodon instances (URLs) on a single server? Like for example, could I have a Mastodon site at example1.com and another Mastodon site at example2.com running on the same server? I found this link but I'm not sure if it's the same thing.

https://github.com/tootsuite/documentation/blob/master/Running-Mastodon/Serving_a_different_domain.md

**Comment by Gargron** [comment]
Yes, you can. Different DB_NAME, different REDIS_NAMESPACE, different PORT for web/streaming services, multiple server definitions in nginx.

**Comment by alexnya** [comment]
Can you over it in more detail or is there a link that explains this?

**Comment by Gargron** [comment]
You can run the same code multiple times, but you'll need to adjust the configuration for each separate deployment so they don't overlap the databases or ports. Only one process can listen on a port. So you'd have mastodon1-web.service with PORT=3000 and mastodon2-web.service with PORT=3010 for example. There is no documentation specifically for this.

**Comment by alexnya** [comment]
So here's what I did so far to create a second Mastodon instance. I didn't rerun the code where you have to install a package, but rather I looked at the parts where I have to configure a database or port like you said. According to the production guide, I created two virtual hosts (mastodon and mastodon_2). I have a couple of questions:

1. Do I have to create another user (database and linux user)? 

2. Would I need to create another .env.production in home/mastodon/live to change the database and redis_host for the second Mastodon instance? If so how should I name it?

3. For the three different Mastodon services, how should I name them? Is what you suggest (mastodon1-web.service and mastodon2-web.service) the correct way to name them? Which ports should I use for each one?

**Comment by taicv** [comment]
I found this link
https://nfgworld.com/multiple-mastodon-instances-on-a-single-server/
Hope this helps!

**Comment by bpawel-bclub** [comment]
Re-opening this with a different question: 
– how much of a server resources would you need to run two instances, both in docker containers?
Would 3 vCPU / 4GB RAM be fine with that? for small public instances initially, perhaps scaling up later.
